tests/Bridge/Doctrine/Orm/Filter/BooleanFilterTest.php

```diff
@@ -0,0 +1,250 @@
+     *  - properties to test. Keys are the property name. If the value is true, the filter should work on the property,
+     *    otherwise not.
+     *  - expected DQL query.
+     *
+     * @return array
+     */
+    public function filterProvider()
+    {
+        return [
+            // test with true value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => 'true',
+
+                ],
+                sprintf('SELECT o FROM %s o where o.dummyBoolean = :dummyboolean_123456abcdefg', Dummy::class),
+            ],
+            // test with false value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => 'false',
+                ],
+                sprintf('SELECT o FROM %s o where o.dummyBoolean = :dummyboolean_123456abcdefg', Dummy::class),
+            ],
+            // test with non-boolean value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => 'toto',
+                ],
+                sprintf('SELECT o FROM %s o', Dummy::class),
+            ],
+            // test with 0 value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => '0',
+                ],
+                sprintf('SELECT o FROM %s o where o.dummyBoolean = :dummyboolean_123456abcdefg', Dummy::class),
+            ],
+            // test with 1 value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                    'dummyBoolean' => '1',
+                ],
+                sprintf('SELECT o FROM %s o where o.dummyBoolean = :dummyboolean_123456abcdefg', Dummy::class),
+            ],
+            // test with nested properties.
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'relatedDummy.id' => null],
+                ],
+                [
+                    'relatedDummy.id' => '1',
+                ],
+                sprintf('SELECT o FROM %s o left join o.relateddummy relateddummy_123456abcdefg where relateddummy_123456abcdefg.id = :id_123456abcdefg', Dummy::class),
+            ],
+            // test with multiple 1 value
+            [
+                [
+                    'properties' => ['id' => null, 'name' => null, 'dummyBoolean' => null],
+                ],
+                [
+                   'dummyBoolean' => '1',
+                   'name' => '1',
```

Comment:
Why are we testing the `BooleanFilter` with non-boolean fields (e.g. `name`), or is this intentional?